src/persistentstorage.cpp

```diff
@@ -258,3 +265,233 @@ void MysqlPersistentStorage::writelog(const std::string &id, const ObjectValue &
     }
 }
 #endif // HAVE_MYSQL
+
+
+#ifdef SUPPORT_INFLUXDB
+Logger& InfluxdbPersistentStorage::logger_m(Logger::getInstance("InfluxdbPersistentStorage"));
+
+bool existsInArray(const Json::Value &json, const std::string &key)
+{
+    assert(json.isArray());
+    for (int i = 0; i != json.size(); i++)
+    {
+        if (json[i][0].asString() == key)
+            return true;
+    }
+    return false;
+}
+
+bool InfluxdbPersistentStorage::createDB(const std::string &db, std::string &result)
+{
+    std::string query = "q=CREATE DATABASE ";
+    query.append(db);
+    return curlRequest(INFLUXDB_QUERY, db, query, result);
+}
+
+InfluxdbPersistentStorage::InfluxdbPersistentStorage(ticpp::Element* pConfig)
+{
+    uri_m = pConfig->GetAttribute("uri");
+    user_m = pConfig->GetAttribute("user");
+    pass_m = pConfig->GetAttribute("pass");
+    logDb_m = pConfig->GetAttribute("db");
+    persistenceDb_m = pConfig->GetAttribute("persist_db");
+
+    std::string resp;
+
+    if (curlRequest(INFLUXDB_QUERY, logDb_m, "q=SHOW DATABASES", resp))
+    {
+        Json::CharReaderBuilder builder;
+        Json::CharReader * reader = builder.newCharReader();
+        Json::Value root, val;
+        std::string errors;
+
+        if(!reader->parse(resp.c_str(), resp.c_str()+resp.size(), &root, &errors))
+        {
+            logger_m.errorStream() << "couldn't parse JSON! Errors: " << errors << endlog;
+        }
+        else
+        {
+            Json::Value databases;
+            resp.clear();
+            databases = root["results"][0]["series"][0]["values"];
+            if (!existsInArray(databases, logDb_m))
+                if (!createDB(logDb_m, resp))
+                    logger_m.errorStream() << "couldn't create logging database: " << resp << endlog;
+            if (!existsInArray(databases, persistenceDb_m))
+                if (!createDB(persistenceDb_m, resp))
+                    logger_m.errorStream() << "couldn't create persistence database: " << resp << endlog;
+        }
+    }
+}
+
+InfluxdbPersistentStorage::~InfluxdbPersistentStorage()
+{
+}
+
+static size_t _curlWriteCB(void *contents, size_t size, size_t nmemb, void *userp)
+{
+    ((std::string*)userp)->append((char*)contents, size * nmemb);
```

Comment:
As per the C++ standard, `reinterpret_cast<char*>(contents)` should be used here.

Reply:
And `reinterpret_cast<std::string*>(userp)` obviously.